src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -56,9 +56,36 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             catch
             {
                 string message = String.Format(CultureInfo.CurrentCulture,
-                    "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                    "Invalid storage account '{0}'. Please make sure your credentials are correct.",
+                    account.Credentials.AccountName);
                 throw new InvalidOperationException(message);
             }
+
+            if (isPrimaryAccount)
+            {
+                // Primary storage accounts require Queues
+                var queueClient = account.CreateQueueClient();
+                var queue = queueClient.GetQueueReference("name");
+                try
+                {
+                    await queue.ExistsAsync(cancellationToken);
+                    _primaryCredentials = account.Credentials;
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (StorageException exception)
+                {
+                    if (!exception.IsNotFoundQueueNotFound())
```

Comment:
Could this check catch more errors than we want it to? Can you modify it so that it also verifies you're getting the expected 4xx status code so you know the failure is not a 5xx Service Unavailable or something? Looks like currently, if you got a 503, you'd complain about the account.